### PR TITLE
cpp aql collections() and test

### PR DESCRIPTION
**Pull request by baslr**

### CODE REVIEW

File: arangod/Aql/Functions.cpp [hunk 1]
```diff
@@ -1604,6 +1604,74 @@ AqlValue Functions::Sleep(arangodb::aql::Query* query,
   return AqlValue(AqlValueHintNull());
 }
 
+#include "GeneralServer/AuthenticationFeature.h"
```
Comment (jsteemann) [inline arangod/Aql/Functions.cpp]: please move includes to the top of the file

File: arangod/Aql/Functions.cpp [hunk 2]
```diff
@@ -1604,6 +1604,74 @@ AqlValue Functions::Sleep(arangodb::aql::Query* query,
...
+    cleanup = [&colls]() {
+      for (auto& it : colls) {
+        if (it != nullptr) {
+          delete it;
+        }
+      }
+    };
+    colls = GetCollectionsCluster(vocbase);
+  } else {
+    colls = vocbase->collections(false);
```
Comment (jsteemann) [inline arangod/Aql/Functions.cpp]: `cleanup = []() {};` is missing!

File: arangod/Aql/Functions.cpp [hunk 3]
```diff
@@ -1604,6 +1604,74 @@ AqlValue Functions::Sleep(arangodb::aql::Query* query,
...
+        continue;
+      }
+    }
+
+    builder->openObject();
+    builder->add("_id", VPackValue(collection->cid_as_string()));
+    builder->add("name", VPackValue(collection->name()));
+    builder->close();
+
+    collection = nullptr;
```
Comment (jsteemann) [inline arangod/Aql/Functions.cpp]: this will produce a memleak. please delete this line.

File: js/server/tests/aql/aql-functions-misc.js [hunk 4]
```diff
@@ -646,9 +645,17 @@ function ahuacatlCollectionCountTestSuite () {
...
+   },
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief test COLLECTIONS()
+////////////////////////////////////////////////////////////////////////////////
+
+    testCollections : function () {
+      assertEqual(db._collections().map((col) => {return {name:col.name(),_id:''+col._id}}), getQueryResults('RETURN NOOPT(COLLECTIONS())')[0]);
+      assertEqual(db._collections().map((col) => {return {name:col.name(),_id:''+col._id}}), getQueryResults('RETURN NOOPT(V8(COLLECTIONS()))')[0]);
     }
```
Comment (jsteemann) [inline js/server/tests/aql/aql-functions-misc.js]: isn't col._id a string anyway?

File: arangod/Aql/Functions.cpp (hunk 1 above)
Comment (baslr) [reply arangod/Aql/Functions.cpp]: done

File: arangod/Aql/Functions.cpp (hunk 2 above)
Comment (baslr) [reply arangod/Aql/Functions.cpp]: done

File: arangod/Aql/Functions.cpp (hunk 3 above)
Comment (baslr) [reply arangod/Aql/Functions.cpp]: done

File: js/server/tests/aql/aql-functions-misc.js (hunk 4 above)
Comment (baslr) [reply js/server/tests/aql/aql-functions-misc.js]: true true

**Pull request merged**